### PR TITLE
create ReadOnlySpan<char> from string

### PR DESCRIPTION
**Pull request by adamsitnik**
@KrzysztofCwalina @jkotas the possibility of creating readonly span from string was missing, so I implemented it

### CODE REVIEW

File: src/mscorlib/src/System/ReadOnlySpan.cs [hunk 1]
```diff
@@ -203,4 +203,62 @@ public bool TryCopyTo(Span<T> destination)
...
+
+    public static class ReadOnlySpanExtensions
+    {
+        /// <summary>
+        /// Creates a new readonly span over the portion of the target string.
+        /// </summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is a null
+        /// reference (Nothing in Visual Basic).</exception>
+        public static ReadOnlySpan<char> Slice(this string text)
```
Comment (jkotas) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: I am not sure that `Slice` is a good name for this method. @KrzysztofCwalina ?

File: src/mscorlib/src/System/ReadOnlySpan.cs [hunk 2]
```diff
@@ -203,4 +203,62 @@ public bool TryCopyTo(Span<T> destination)
...
+        {
+            if (text == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
+
+            return new ReadOnlySpan<char>(ref JitHelpers.GetStringData(text), text.Length);
```
Comment (jkotas) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: You should not need a new JitHelpers method. Adding a new method like `ref char GetFirstCharRef() { return ref m_firstChar; }` should be enough.

File: src/mscorlib/src/System/ReadOnlySpan.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: Would it be better to expose this as an implicit cast?

Comment (KrzysztofCwalina) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: Ah, this is not on String. Well, in this case case won't work.

Comment (adamsitnik) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: Maybe I should add the implicit cast operator and keep the Slice method?

File: src/mscorlib/src/System/ReadOnlySpan.cs (hunk 2 above)
Comment (adamsitnik) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: @jkotas this has simplified things a lot! I pushed an update

File: src/mscorlib/src/System/ReadOnlySpan.cs (hunk 1 above)
Comment (adamsitnik) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: @KrzysztofCwalina I can not add implicit cast. I get compiler error `User-defined conversion must convert to or from the enclosing type` (if I add implicit cast to ReadOnlySpan of `T` to cast to ROS of `char`)

Comment (KrzysztofCwalina) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: Yeah, that's what I meant. In this case, I am fine with Slice(), i.e I have no better naming suggestions.

File: src/mscorlib/model.xml [hunk 3]
```diff
@@ -12100,5 +12100,10 @@
       <Member Name="NonPortableCast&lt;TFrom,TTo&gt;(System.Span&lt;TFrom&gt;)" />
       <Member Name="NonPortableCast&lt;TFrom,TTo&gt;(System.ReadOnlySpan&lt;TFrom&gt;)" />
     </Type>
+    <Type Name="System.ReadOnlySpanExtensions">
```
Comment (jkotas) [inline src/mscorlib/model.xml]: It may be useful to add this with `Condition="FEATURE_SPAN_OF_T` like the other span apis.

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by KrzysztofCwalina** [comment]
looks good to me too.

File: src/mscorlib/model.xml (hunk 3 above)
Comment (adamsitnik) [inline src/mscorlib/model.xml]: @jkotas  I have pushed commit with the condition

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**